command/arguments/view.go

```diff
@@ -0,0 +1,35 @@
+package arguments
+
+// View represents the global command-line arguments which configure the view.
+type View struct {
+	// NoColor is used to disable the use of terminal color codes in all
+	// output.
+	NoColor bool
+
+	// CompactWarnings is used to coalesce duplicate warnings, to reduce the
+	// level of noise when multiple instances of the same warning are raised
+	// for a configuration.
+	CompactWarnings bool
+}
+
+// ParseView processes CLI arguments, returning a View value and a
+// possibly-modified slice of arguments. If any of the supported flags are
+// found, they will be removed from the slice.
+func ParseView(args []string) (*View, []string) {
+	common := &View{}
+
+	i := 0
+	for _, v := range args {
+		switch v {
+		case "-no-color":
+			common.NoColor = true
+		case "-compact-warnings":
+			common.CompactWarnings = true
+		default:
+			// copy and increment index
+			args[i] = v
+			i++
+		}
+	}
+	return common, args[:i]
```

Comment:
I see what you're doing here with the return slice, but I had to stare at it for a few moments first - it's not wrong (or bad), but it's also not intuitive to me. Could you maybe update the comment above, or add something here, that expands a bit?

Reply:
Ah, yes. This filter-the-slice loop confuses me every time I see it too, so I've added a bit of commentary. Thanks!